core/src/main/java/org/apache/calcite/plan/SubstitutionVisitor.java

```diff
@@ -593,6 +593,23 @@ assert equalType(
     return substitutions;
   }
 
+  /**
+   * Equivalence checking for row types, but except for the field names.
+   */
+  private Boolean rowTypesAreEquivalent(
```

Comment:
Shouldn't this method return `boolean` instead of `Boolean` ?

Reply:
Sorry, updated :)